### DISCUSSION TITLE
"Error when loading first segment" in only certain videos within a website

### DISCUSSION THREAD

**Discussion by bifasp** [aclap-dev/video-downloadhelper#29] [Help and Q&A] [answered]
### Your question

Using Firefox 122.0 on macOS Monterey 12.7.3 and Video DownloadHelper 8.1.4.0 with VdhCoApp 2.0.10.

I use to download videos from a private website for about 2 years now. Since 15 days I'm struggling to download the latest videos only. All old videos can be downloaded without issues and all of them are hosted in the same server.

As sharing the public url won't be possible, I'm pasting here the error I'm getting in VDH. I tried to use a Windows machine with the same Firefox version and VDH beta 8.2 with the same results. Any help will be appreciated as I'm not a power user and never used ffmpeg via command line.

Here's the error returned by VDH:

GrabInfo: Cannot get info from https://iframe.mediadelivery.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/842x480/video.drm?contextId=c9f18310-f0f6-4e42-9d0f-d28fee0b4a54:Exit code: 1 ffprobe version n6.0 Copyright (c) 2007-2023 the FFmpeg developers built with Apple clang version 15.0.0 (clang-1500.0.40.1) configuration: --extra-libs= --extra-ldflags= --extra-ldexeflags= --pkg-config-flags=--static --disable-autodetect --prefix=/Users/paul/git/ffmpeg-static-builder/dist/mac/x86_64//ffmpeg --enable-version3 --pkg-config=pkg-config --enable-runtime-cpudetect --enable-pthreads --disable-w32threads --enable-libtheora --enable-libvpx --enable-libvorbis --disable-indev=sndio --disable-outdev=sndio --enable-neon --enable-libopenjpeg --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-gpl --enable-openssl --enable-libvo-amrwbenc --enable-libopus --enable-libx264 --enable-libx265 --enable-libaom --enable-libwebp --enable-libxvid --enable-libmp3lame --enable-zlib --enable-ffprobe --enable-pic --disable-doc libavutil 58. 2.100 / 58. 2.100 libavcodec 60. 3.100 / 60. 3.100 libavformat 60. 3.100 / 60. 3.100 libavdevice 60. 1.100 / 60. 1.100 libavfilter 9. 3.100 / 9. 3.100 libswscale 7. 1.100 / 7. 1.100 libswresample 4. 10.100 / 4. 10.100 libpostproc 57. 1.100 / 57. 1.100 [hls @ 0x7fd504f0bf40] Skip ('#EXT-X-VERSION:3') [hls @ 0x7fd504f0bf40] Opening 'https://video-987.mediadelivery.net/.drm/c9f18310-f0f6-4e42-9d0f-d28fee0b4a54/842x480.drmkey?v=0' for reading [hls @ 0x7fd504f0bf40] Opening 'crypto+https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/842x480/video0.ts?v=0&resolution=842x480&server=987&contextId=c9f18310-f0f6-4e42-9d0f-d28fee0b4a54' for reading [hls @ 0x7fd504f0bf40] Opening 'crypto+https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/842x480/video1.ts?v=0&resolution=842x480&server=987&contextId=c9f18310-f0f6-4e42-9d0f-d28fee0b4a54' for reading [hls @ 0x7fd504f0bf40] Opening 'crypto+https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/842x480/video2.ts?v=0&resolution=842x480&server=987&contextId=c9f18310-f0f6-4e42-9d0f-d28fee0b4a54' for reading [hls @ 0x7fd504f0bf40] Error when loading first segment 'https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/842x480/video0.ts?v=0&resolution=842x480&server=987&contextId=c9f18310-f0f6-4e42-9d0f-d28fee0b4a54' https://iframe.mediadelivery.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/842x480/video.drm?contextId=c9f18310-f0f6-4e42-9d0f-d28fee0b4a54: Invalid data found when processing input
moz-extension://3df748d8-87bd-4500-88d6-3ed55575127f/background/main.js:31
cn</grabInfo/<@moz-extension://3df748d8-87bd-4500-88d6-3ed55575127f/background/main.js:31:88385

### Addon & CoApp version and details

Firefox 122.0 on macOS Monterey 12.7.3 and Video DownloadHelper 8.1.4.0 with VdhCoApp 2.0.10.

**Comment by paulrouget** [comment]
Hey, do you mind testing with a more recent version of the CoApp: https://www.downloadhelper.net/w/CoApp-Installation

**Reply by bifasp** [reply]
Thanks for the suggestion. I used the latest version you proposed. Basically the same error was returned. The difference now is that it was possible to detect a file with higher resolution. But the download process won't start.

GrabInfo: Cannot get info from https://iframe.mediadelivery.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video.drm?contextId=bae8387c-5148-471b-8e7b-bc373a3d233a:Exit code: 1 ffprobe version N-113507-gf6b7b473d4-tessus https://evermeet.cx/ffmpeg/ Copyright (c) 2007-2024 the FFmpeg developers built with Apple clang version 11.0.0 (clang-1100.0.33.17) configuration: --cc=/usr/bin/clang --prefix=/opt/ffmpeg --extra-version=tessus --enable-avisynth --enable-fontconfig --enable-gpl --enable-libaom --enable-libass --enable-libbluray --enable-libdav1d --enable-libfreetype --enable-libgsm --enable-libmodplug --enable-libmp3lame --enable-libmysofa --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenh264 --enable-libopenjpeg --enable-libopus --enable-librubberband --enable-libshine --enable-libsnappy --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libtwolame --enable-libvidstab --enable-libvmaf --enable-libvo-amrwbenc --enable-libvorbis --enable-libvpx --enable-libwebp --enable-libx264 --enable-libx265 --enable-libxavs --enable-libxml2 --enable-libxvid --enable-libzimg --enable-libzmq --enable-libzvbi --enable-version3 --pkg-config-flags=--static --disable-ffplay libavutil 58. 36.101 / 58. 36.101 libavcodec 60. 38.100 / 60. 38.100 libavformat 60. 20.100 / 60. 20.100 libavdevice 60. 4.100 / 60. 4.100 libavfilter 9. 17.100 / 9. 17.100 libswscale 7. 6.100 / 7. 6.100 libswresample 4. 13.100 / 4. 13.100 libpostproc 57. 4.100 / 57. 4.100 mime type is not rfc8216 compliant [hls @ 0x7fa488f133c0] Skip ('#EXT-X-VERSION:3') [hls @ 0x7fa488f133c0] Opening 'https://video-987.mediadelivery.net/.drm/bae8387c-5148-471b-8e7b-bc373a3d233a/1920x1080.drmkey?v=0' for reading [hls @ 0x7fa488f133c0] Opening 'crypto+https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video0.ts?v=0&resolution=1920x1080&server=987&contextId=bae8387c-5148-471b-8e7b-bc373a3d233a' for reading [hls @ 0x7fa488f133c0] Opening 'crypto+https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video1.ts?v=0&resolution=1920x1080&server=987&contextId=bae8387c-5148-471b-8e7b-bc373a3d233a' for reading [hls @ 0x7fa488f133c0] Error when loading first segment 'https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video0.ts?v=0&resolution=1920x1080&server=987&contextId=bae8387c-5148-471b-8e7b-bc373a3d233a' https://iframe.mediadelivery.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video.drm?contextId=bae8387c-5148-471b-8e7b-bc373a3d233a: Invalid data found when processing input

moz-extension://3df748d8-87bd-4500-88d6-3ed55575127f/background/main.js:31
cn</grabInfo/<@moz-extension://3df748d8-87bd-4500-88d6-3ed55575127f/background/main.js:31:88385

**Comment by GrampaWildWilly** [comment]
I was about to suggest the same thing Paul did.  Also try the VDH you can find here: https://github.com/aclap-dev/video-downloadhelper/wiki/Beta-versions.

Meanwhile, I note the appearance of the string DRM in the URLs you posted.  Digital Rights Management is a technique web sites use to block downloading.  DRM is supposed to be a way to protect copyrighted material from unauthorized use.  If this web site has suddenly decided to put its content behind DRM, you're flat out of luck.  Doing such a thing is not unheard of.  I've hit it on at least one site VDH used to work well on.  That's life.

Now, there is a way to figure out whether something is protected by DRM.  Go into the Firefox settings & find the checkbox where it allows you to play DRM-controlled content.  Uncheck the box, then reload your web page.  You might have to also click PLAY on the web page.  If you get a message at the top of the Firefox window about DRM, then your goose is cooked.  They've put DRM in place & you're stuck.  You could probably get the content using a screen recorder.  That's not as good as downloading, but it might be your only recourse.

**Reply by bifasp** [reply]
I went and installed that beta version of VDH. Still get the error even unchecking the DRM box. The video plays nicely with DRM unchecked.

I attach some screenshots here and the latest error returned (not so much details as the previous VDH version): 

SideDownload error:

[hls @ 0x7fc792d04a00] Error when loading first segment 'https://vz-35482fe5-5d0.b-cdn.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video0.ts?v=0&resolution=1920x1080&server=987&contextId=cf7259a9-29de-4d2e-8a5d-9263a4e7bc5f'
[in#0 @ 0x7fc792d048c0] Error opening input: Invalid data found when processing input
Error opening input file https://iframe.mediadelivery.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video.drm?contextId=cf7259a9-29de-4d2e-8a5d-9263a4e7bc5f.
Error opening input files: Invalid data found when processing input


<img width="1131" alt="vdhinfo" src="https://github.com/aclap-dev/video-downloadhelper/assets/158792625/5d9725b5-81a6-4e8d-af71-b5c711d1513d">
<img width="622" alt="drm disabled" src="https://github.com/aclap-dev/video-downloadhelper/assets/158792625/90283321-4638-4e21-beab-9b8126fc0401">

**Reply by GrampaWildWilly** [reply]
Most puzzling.

We've had similar baffling results that turned out to be related to antivirus & firewall issues.  You might want to read this:

https://github.com/aclap-dev/video-downloadhelper/discussions/14

What antivirus & firewall are you running?

**Reply by paulrouget** [reply, marked as answer]
> What antivirus & firewall are you running?

This is MacOS, so most likely not an issue.

> Error opening input: Invalid data found when processing input
> [video.drm?contextId=cf7259a9-29de-4d2e-8a5d-9263a4e7bc5f](https://iframe.mediadelivery.net/e7686b8c-cee6-43aa-ae34-7f6525622a79/1920x1080/video.drm?contextId=cf7259a9-29de-4d2e-8a5d-9263a4e7bc5f).

This is DRM content.

So if Firefox can still play, that means VDH offers DRM content while Firefox use another type of content. That means VDH is filtering out a non-DRM cotnent.

@bifasp assuming you're using v8.2, go to `Settings > More > Video Qualities`, and uncheck ` Ignore low quality videos`.

Reload the page, and tell me if you see more potential downloads. One must be DRM-free.

**Reply by GrampaWildWilly** [reply]
Learning, learning . . .

Don't Mac users go in for antivirus & firewall software?

And both DRM & non-DRM versions of the same video?  I never imagined that was even possible.  Why would a web site do something like that?  Just to thwart downloaders?  That is evil . . .

**Reply by paulrouget** [reply]
> Don't Mac users go in for antivirus & firewall software?

Very uncommon.

> And both DRM & non-DRM versions of the same video? I never imagined that was even possible. Why would a web site do something like that?

That's the only explanation of why this user could see the video with DRM disabled.

**Reply by bifasp** [reply]
Sorry for the late reply @paulrouget and @GrampaWildWilly . I do use Norton 360 on the Mac. Not because of virus but because I use its VPN service on the Mac while the rest of the family using Windows use the whole suite (I have a family plan).
Anyway, I disabled both Norton on the Mac and the TrendMicro content filter which is built in my main internet router. Guess what? Same result.
I also tried going to Settings > More > Video Qualities, and unchecking  Ignore low quality videos. For a moment, I thought it would work as the progress bar showed 1% download. Then briefly after that, the error occurred again. :-(

**Reply by bifasp** [reply]
AN UPDATE @paulrouget and @GrampaWildWilly : for reasons I don't understand, VDH was displaying 3 or 4 versions of the file. When I selected the last one, IT WORKED (DRM free?). I know I tried this before and it never worked so I believe that the Ignore Low Quality Videos thing did the trick even with Norton and the TrendMicro enabled.
So THANK YOU VERY MUCH GUYS!!!!!

**Reply by paulrouget** [reply]
Thank you for testing out. This then confirms my theory that some DRM-content still provide non-DRM streams for fallback. But then our "Low Quality" filter kicks in and remove the non-DRM source.

I'll need to find a smart way to deal with that.
